src/watcher.ts

```diff
@@ -233,7 +237,71 @@ export async function unsubscribe(endpoint: string, creds: any, subId: string) {
   }
 }
 
-export function filterSubs(subs: any[], filter: IListFilter) {
+export async function testAddressAgainstPendingBloom (
+  endpoint: string,
+  creds: any,
+  address: string
+) {
+  const url = new URL(endpoint + '/subscription/pending/address/' + address)
+
+  const request = aws4.sign(
+    {
+      host: url.host,
+      url: url.href,
+      method: 'GET',
+      path: `${url.pathname}${url.search}`
+    },
+    creds
+  )
+
+  const reqConfig = {
+    method: request.method,
+    url: request.url,
+    headers: request.headers
+  }
+
+  try {
+    const response = await axios.request(reqConfig)
+    return response.data
+  } catch (err) {
+    console.error('Error adding address to pending bloom', err)
+    throw err
+  }
+}
```

Comment:
❔ I see this pattern is repeated a lot in this module; is it useful to refactor into a separate function?

Reply:
yes! issue created